runtime/src/main/java/org/corfudb/protocols/wireprotocol/NettyCorfuMessageEncoder.java

```diff
@@ -12,12 +12,26 @@
 @Slf4j
 public class NettyCorfuMessageEncoder extends MessageToByteEncoder<CorfuMsg> {
 
+
+    Integer max = 0;
+
     @Override
     protected void encode(ChannelHandlerContext channelHandlerContext,
                           CorfuMsg corfuMsg,
                           ByteBuf byteBuf) throws Exception {
         try {
             corfuMsg.serialize(byteBuf);
+            if(log.isDebugEnabled()) {
+                byteBuf.readableBytes();
+                synchronized (max) {
+                    int readableBytes = byteBuf.readableBytes();
+                    if (max < readableBytes) {
+                        max = readableBytes;
```

Comment:
On this assignment, the synchronized lock will be lost as it is acquired on the Integer object. Right? You have to use a final object for synchrony,